interop/authzen-search-demo/app/routes/resource.tsx

```diff
@@ -0,0 +1,292 @@
+import type { Route } from "./+types/resource";
+import {
+  Card,
+  CardContent,
+  CardDescription,
+  CardHeader,
+  CardTitle,
+} from "~/components/ui/card";
+
+import {
+  Table,
+  TableBody,
+  TableCell,
+  TableHead,
+  TableHeader,
+  TableRow,
+} from "~/components/ui/table";
+import { Code } from "lucide-react";
+import { Tabs, TabsContent, TabsList, TabsTrigger } from "~/components/ui/tabs";
+
+import {
+  Select,
+  SelectContent,
+  SelectItem,
+  SelectTrigger,
+  SelectValue,
+} from "~/components/ui/select";
+import { Form, useNavigation } from "react-router";
+import { Button } from "~/components/ui/button";
+import { pdpCookie } from "~/cookies.server";
+import { Prism as SyntaxHighlighter } from "react-syntax-highlighter";
+import { subjects } from "~/data/subjects.server";
+import { pdps } from "~/data/pdps.server";
+import vscDarkPlus from "react-syntax-highlighter/dist/cjs/styles/prism/vsc-dark-plus";
+
+export function meta({}: Route.MetaArgs) {
+  return [{ title: "AuthZEN Search API - Resource Search" }];
+}
+
+export async function loader({}: Route.LoaderArgs) {
+  return {
+    subjects,
+  };
+}
+
+export async function action({ request }: Route.ActionArgs) {
+  const cookieHeader = request.headers.get("Cookie");
+  const cookie = (await pdpCookie.parse(cookieHeader)) || {};
+  if (!cookie.selectedPdp) {
+    return {
+      error: "No PDP selected",
+    };
+  }
+
+  const pdp = pdps.find((pdp) => pdp.name === cookie.selectedPdp);
+  if (!pdp) {
+    return {
+      error: "PDP not found",
+    };
+  }
+
+  const formData = await request.formData();
+  const subject = formData.get("subject");
+  const resource = formData.get("resource");
+  const action = formData.get("action");
+  const nextToken = formData.get("next_token") ?? "";
+
+  if (!subject || subjects.find((s) => s.id === subject) === undefined) {
+    return {
+      error: "Invalid subject selected",
+    };
+  }
+
+  const authZENRequest = {
+    subject: {
+      type: "user",
+      id: subject,
+    },
+    action: {
+      name: action,
+    },
+    resource: {
+      type: resource,
+    },
+    context: {},
+    page: {
```

Comment:
I think we should leave the paging out of this interop demo (as we agreed in the previous meeting)